### PR TITLE
feat: ui.link

### LINKED ISSUE #860: docs: ui.link

### LINKED ISSUE #977: ui.link
Will also include https://github.com/deephaven/deephaven-plugins/issues/860

### PR DESCRIPTION
**Pull request by AkshatJawne**
Closes #977 and Closes #860

**Need to clarify:**
- Omitted implementation of `download`, `ping`, `href_lang`, `referrer_policy`, `router_options` since they do not appear to be working in the spectrum implementation, and they do not seem super needed (given that this is meant to serve as a basic link component)
    -  If we want to implement said props, I can implement special logic for them

### CODE REVIEW

File: plugins/ui/docs/components/link.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+## API Reference
+
+```{eval-rst}
+.. dhautofunction:: deephaven.ui.link
+```
+
+
+
+
```
Comment (dsmmcken) [inline plugins/ui/docs/components/link.md:105]: extra new lines

File: plugins/ui/docs/components/link.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+The link component accept other components, such as `text` and `icon`, as children.
+
+```python
+from deephaven import ui
+
+
+@ui.component
+def ui_link_content_examples():
+    return [
+        ui.link(ui.icon("github_alt"), href="https://github.com/deephaven"),
```
Comment (dsmmcken) [inline plugins/ui/docs/components/link.md:25]:
```suggestion
        ui.link(ui.icon("github"), href="https://github.com/deephaven"),
```

File: plugins/ui/docs/components/link.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+from deephaven import ui
+
+
+@ui.component
+def ui_link_variant_examples():
+    return [
+        ui.link("Deephaven", href="https://deephaven.io/", variant="primary"),
+        ui.link(
+            "Contact the team",
+            href="https://deephaven.io/landing-page/one-on-one/",
```
Comment (dsmmcken) [inline plugins/ui/docs/components/link.md:48]:
```suggestion
            href="https://deephaven.io/contact",
```

File: plugins/ui/docs/components/link.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+```python
+from deephaven import ui
+
+
+my_link_over_background_example = ui.view(
+    ui.link(
+        "Learn more about pandas here!",
+        href="https://en.wikipedia.org/wiki/Giant_panda",
+        variant="over_background",
```
Comment (dsmmcken) [inline plugins/ui/docs/components/link.md:69]: I don't think this is working, text should display white but it's not.

File: plugins/ui/docs/components/link.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+my_link_is_quiet_example = ui.view(
+    ui.action_group(
+        ui.item("Add"),
+        ui.item("Edit"),
+        ui.item("Delete"),
+        is_quiet=True,
+    ),
+    ui.text("Confused about these options? "),
+    ui.link("Learn more here", is_quiet=True),
+)
```
Comment (dsmmcken) [inline plugins/ui/docs/components/link.md:93]:
```suggestion
from deephaven import ui


my_link_is_quiet_example = ui.text(
    "You can ", ui.link("use quiet", is_quiet=True), " links inline."
)
```

**Review by dsmmcken — CHANGES_REQUESTED** [review]

File: plugins/ui/src/deephaven/ui/components/link.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+def link(
+    *children: Any,
+    variant: LinkVariant | None = "primary",
+    is_quiet: bool | None = None,
+    auto_focus: bool | None = None,
+    href: str | None = None,
+    target: Target | None = None,
+    rel: str | None = None,
+    # omitted download, ping, href_lang, referrer_policy, router_options
```
Comment (dsmmcken) [inline plugins/ui/src/deephaven/ui/components/link.py:28]: download, ping, href_lang, referrer_policy should be supported and are native browser behaviours. 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attributes

File: plugins/ui/src/js/src/elements/Link.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+} from '@deephaven/components';
+
+export function Link(props: DHCLinkProps): JSX.Element {
+  return (
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    <DHCLink {...props} />
+  );
+}
+
+Link.displayName = 'Link';
```
Comment (mofojed) [inline plugins/ui/src/js/src/elements/Link.tsx:14]: You're not modifying any of the props here, so I don't think you need to make this wrapper and in WidgetUtils you can just import straight from `@deephaven/components`.

**Review by mofojed — CHANGES_REQUESTED** [review]

File: plugins/ui/docs/components/link.md (hunk 1 above)
Comment (dsmmcken) [reply plugins/ui/docs/components/link.md:98]: still there

**Review by mofojed — DISMISSED** [review]

File: plugins/ui/docs/components/link.md (hunk 1 above)
Comment (AkshatJawne) [reply plugins/ui/docs/components/link.md]: Should be good now.

**Review by margaretkennedy — APPROVED** [review]

**Review by dsmmcken — APPROVED** [review]

**Pull request merged**